avm-transpiler/src/transpile.rs

```diff
@@ -0,0 +1,289 @@
+                });
+            }
+            BrilligOpcode::Store {
+                destination_pointer,
+                source,
+            } => {
+                // INDIRECT dstOffset operand (bit 1 set high)
+                avm_instrs.push(AvmInstruction {
+                    opcode: AvmOpcode::MOV,
+                    indirect: Some(FIRST_OPERAND_INDIRECT), // indirect dstOffset operand
+                    operands: vec![
+                        AvmOperand::U32 {
+                            value: source.to_usize() as u32,
+                        },
+                        AvmOperand::U32 {
+                            value: destination_pointer.to_usize() as u32,
+                        },
+                    ],
+                    ..Default::default()
+                });
+            }
+            BrilligOpcode::Call { location } => {
+                let avm_loc = brillig_pcs_to_avm_pcs[*location];
+                avm_instrs.push(AvmInstruction {
+                    opcode: AvmOpcode::INTERNALCALL,
+                    operands: vec![AvmOperand::U32 {
+                        value: avm_loc as u32,
+                    }],
+                    ..Default::default()
+                });
+            }
+            BrilligOpcode::Return {} => avm_instrs.push(AvmInstruction {
+                opcode: AvmOpcode::INTERNALRETURN,
+                ..Default::default()
+            }),
+            BrilligOpcode::Stop { return_data_offset, return_data_size } => {
+                avm_instrs.push(AvmInstruction {
+                    opcode: AvmOpcode::RETURN,
+                    operands: vec![
+                        AvmOperand::U32 { value: *return_data_offset as u32},
+                        AvmOperand::U32 { value: *return_data_size as u32},
+                    ],
+                    ..Default::default()
+                });
+            }
+            BrilligOpcode::Trap { /*return_data_offset, return_data_size*/ } => {
+                // TODO: Trap should support return data
+                avm_instrs.push(AvmInstruction {
+                    opcode: AvmOpcode::REVERT,
+                    operands: vec![
+                        //AvmOperand::U32 { value: *return_data_offset as u32},
+                        //AvmOperand::U32 { value: *return_data_size as u32},
+                        AvmOperand::U32 { value: 0},
+                        AvmOperand::U32 { value: 0},
+                    ],
+                    ..Default::default()
+                });
+            }
+            _ => panic!(
+                "Transpiler doesn't know how to process {:?} brillig instruction",
+                brillig_instr
+            ),
+        }
+    }
+
+    dbg_print_avm_program(&avm_instrs);
+
+    // Constructing bytecode from instructions
+    let mut bytecode = Vec::new();
+    for i in 0..avm_instrs.len() {
+        let mut instr_bytes = avm_instrs[i].to_bytes();
+        bytecode.append(&mut instr_bytes);
```

Comment:
mut can be removed if we use `bytecode.extend_from_slice(&instr_bytes)`

Reply:
i'm a rust noob, thanks